src/transformStateWithClones.js

```diff
@@ -7,7 +7,52 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = Object.assign({}, state);
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ........ } = action;
    
    switch (type) {
```

Reply:
Fixed